utils/queue/priorityQueue.go

```diff
@@ -0,0 +1,84 @@
+package queue
+
+import (
+	"container/heap"
+	"time"
+)
+
+// Item stores the attributes which will be pushed to the priority queue..
+type Item[T any] struct {
+	Value     T
+	Priority  int
+	timeStamp int64
+	index     int
+}
+
+// PriorityQueue ....
+type PriorityQueue[T any] []*Item[T]
+
+// Len: Size of the priority queue . Used to satisfy the heap interface...
```

Comment:
I got some linting stuff out of this file. Beside JetBrains complaining for `:` after `Len` saying:
> Comment should have the following format 'Len ...' (with an optional leading article) 

On a more serious note, some methods of `PriorityQueue` have method receivers, some do not. Is that intended?

Reply:
Yes, using pointer and non-pointer receivers per case is intended and not any different than [Go's own proposed implementation](https://pkg.go.dev/container/heap#example-package-PriorityQueue) of a priority queue.